packages/docusaurus/src/commands/deploy.ts

```diff
@@ -97,18 +97,23 @@ export default async function deploy(
     (projectName.indexOf('.github.io') !== -1 ? 'master' : 'gh-pages');
   console.log(`${chalk.cyan('deploymentBranch:')} ${deploymentBranch}`);
 
+  const useSSH = process.env.USE_SSH;
+
   const githubHost =
     process.env.GITHUB_HOST || siteConfig.githubHost || 'github.com';
+  const githubPort =
+    process.env.GITHUB_PORT || siteConfig.githubPort || (useSSH && useSSH.toLowerCase() === 'true'
+      ? '22'
+      : '433');
 
-  const useSSH = process.env.USE_SSH;
   const gitPass: string | undefined = process.env.GIT_PASS;
   let gitCredentials = `${gitUser}`;
   if (gitPass) {
     gitCredentials = `${gitCredentials}:${gitPass}`;
   }
 
-  const sshRemoteBranch: string = `git@${githubHost}:${organizationName}/${projectName}.git`;
-  const nonSshRemoteBranch: string = `https://${gitCredentials}@${githubHost}/${organizationName}/${projectName}.git`;
+  const sshRemoteBranch: string = `git@${githubHost}:${githubPort}/${organizationName}/${projectName}.git`;
```

Comment:
what is the impact of this change? 
It looks better to me but it is different from the pattern `${githubHost}:${organizationName}` we used before, do you have any doc so that I can double-check that?

Reply:
if we look at what is suggested by the Github UI:

![image](https://user-images.githubusercontent.com/749374/113280771-0987e880-92e5-11eb-9b88-c40e667778d1.png)

Reply:
https://tools.ietf.org/id/draft-salowey-secsh-uri-00.html

I'm trying to follow this pattern and then make it reusable out of github.
```
ssh://git@<HOST>:<PORT>/<REPO_PATH>
```

I'm working with Bitbucket/Stash and I need to define the port. 

```
git@github.com:facebook/docusaurus.git == ssh://git@github.com:22/facebook/docusaurus.git
```